walkthroughs/howto-http-retries/deploy.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -z $AWS_ACCOUNT_ID ]; then
+    echo "AWS_ACCOUNT_ID environment variable is not set."
+    exit 1
+fi
+
+if [ -z $AWS_DEFAULT_REGION ]; then
+    echo "AWS_DEFAULT_REGION environment variable is not set."
+    exit 1
+fi
+
+if [ -z $ENVOY_IMAGE ]; then
+    echo "ENVOY_IMAGE environment variable is not set to App Mesh Envoy, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html"
+    exit 1
+fi
+
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+PROJECT_NAME="howto-http-retries"
+ECR_IMAGE_PREFIX=${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/${PROJECT_NAME}
+
+deploy_images() {
+    for app in colorapp feapp; do
+        aws ecr describe-repositories --repository-name $PROJECT_NAME/$app >/dev/null 2>&1 || aws ecr create-repository --repository-name $PROJECT_NAME/$app
+        docker build -t ${ECR_IMAGE_PREFIX}/${app} ${DIR}/${app}
+        $(aws ecr get-login --no-include-email)
+        docker push ${ECR_IMAGE_PREFIX}/${app}
+    done
+}
+
+deploy_infra() {
+    stack_name="${PROJECT_NAME}-infra"
+    aws cloudformation deploy \
+        --no-fail-on-empty-changeset \
+        --stack-name $stack_name\
+        --template-file "${DIR}/infra.yaml" \
+        --capabilities CAPABILITY_IAM \
+        --parameter-overrides "ProjectName=${PROJECT_NAME}"
+}
+
+deploy_app() {
+    aws cloudformation deploy \
+        --no-fail-on-empty-changeset \
+        --stack-name "${PROJECT_NAME}-app" \
+        --template-file "${DIR}/app.yaml" \
+        --capabilities CAPABILITY_IAM \
+        --parameter-overrides "ProjectName=${PROJECT_NAME}" "EnvoyImage=${ENVOY_IMAGE}" "ColorAppImage=${ECR_IMAGE_PREFIX}/colorapp" "FrontAppImage=${ECR_IMAGE_PREFIX}/feapp"
+}
+
+delete_cfn_stack() {
+    stack_name=$1
+    aws cloudformation delete-stack --stack-name $stack_name
+    echo 'Waiting for the stack to be deleted, this may take a few minutes...'
+    aws cloudformation wait stack-delete-complete --stack-name $stack_name
+    echo 'Done'
+}
+
+delete_images() {
+    for app in colorapp feapp; do
+        image_details_text=$(aws ecr describe-images \
+            --repository-name="$PROJECT_NAME/$app" \
+            --query="imageDetails" \
+            --output=json \
+            | jq '.[] | .imageDigest')
+
+        for sha in $image_details_text; do
+            sha=$(sed -e 's/^"//' -e 's/"$//' <<<"$sha")
+
+            echo "delete image $sha..."
+            aws ecr batch-delete-image \
+               --repository-name $PROJECT_NAME/$app \
+               --image-ids imageDigest=$sha
+        done
+
+        echo "deleting repository..."
+        aws ecr delete-repository \
+           --repository-name $PROJECT_NAME/$app
```

Comment:
I forgot to mention in the other one, but AFAIK you can just add `--force`  and it'll delete the images. https://docs.aws.amazon.com/cli/latest/reference/ecr/delete-repository.html

Not blocking.